File: src/main/java/com/bugsnag/android/Logger.java
package com.bugsnag.android;

import android.util.Log;

class Logger {
    private static final String LOG_TAG = "Bugsnag";
    private static boolean enabled = true;

    static void info(String message) {
        if (enabled) {
            Log.i(LOG_TAG, message);
        }
    }

    static void warn(String message) {
        if (enabled) {
            Log.w(LOG_TAG, message);
        }
    }

    static void warn(String message, Throwable e) {
        if (enabled) {
            Log.w(LOG_TAG, message, e);
        }
    }

    static void setEnabled(boolean enabled) {
        Logger.enabled = enabled;
    }

}

File: bugsnag-android-core/src/main/java/com/bugsnag/android/Logger.kt
package com.bugsnag.android

interface Logger {
    fun e(msg: String): Unit = Unit
    fun e(msg: String, throwable: Throwable): Unit = Unit
    fun w(msg: String): Unit = Unit
    fun w(msg: String, throwable: Throwable): Unit = Unit
    fun i(msg: String): Unit = Unit
    fun i(msg: String, throwable: Throwable): Unit = Unit
    fun d(msg: String): Unit = Unit
    fun d(msg: String, throwable: Throwable): Unit = Unit
}
